python/turbine_models/tests/vmfb_comparison.py

```diff
@@ -0,0 +1,183 @@
+import os
+import sys
+import re
+
+from typing import Tuple
+
+os.environ["TORCH_LOGS"] = "dynamic"
+from transformers import AutoTokenizer, AutoModelForCausalLM
+import torch
+from torch.utils import _pytree as pytree
+from shark_turbine.aot import *
+from iree.compiler.ir import Context
+from iree import runtime as ireert
+
+from turbine_models.custom_models import remap_gguf
+import safetensors
+
+from tqdm import tqdm
+
+BATCH_SIZE = 1
+MAX_STEP_SEQ = 4095
+
+
+def torch_token_generator(
+    prompt, hf_model_name: str, hf_auth_token: str, break_on_eos=False
+):
+    tokenizer = AutoTokenizer.from_pretrained(
+        hf_model_name, use_fast=False, use_auth_token=hf_auth_token
+    )
+    model = AutoModelForCausalLM.from_pretrained(
+        hf_model_name, torch_dtype=torch.float, use_auth_token=hf_auth_token
+    )
+
+    initial_input = tokenizer(prompt, return_tensors="pt")
+    input_ids = initial_input.input_ids
+    past_key_values = None
+
+    while True:
+        model_results = model.forward(input_ids, past_key_values=past_key_values)
+        logits = model_results.logits
+        next_token_id = torch.argmax(logits[:, -1, :], dim=1)
+        past_key_values = model_results.past_key_values
+
+        yield next_token_id
+        input_ids = next_token_id.unsqueeze(0)  # Prepare for the next iteration
+
+        if next_token_id.item() == tokenizer.eos_token_id and break_on_eos:
+            break
+
+
+def turbine_token_generator(
+    prompt: str,
+    hf_model_name: str,
+    vmfb_path: str = None,
+    external_weight_file: str = None,
+    hf_auth_token: str = None,
+    break_on_eos: bool = False,
+) -> torch.Tensor:
+    """
+    A generator function for turbine model inference.
+
+    :param prompt: The input prompt for the model.
+    :param hf_model_name: The name of the Hugging Face model.
+    :param vmfb_path: Path to the .vmfb model file.
+    :param external_weight_file: Path to the external weight file (optional).
+    :param hf_auth_token: Hugging Face authorization token (optional).
+    :param break_on_eos: Whether to break the loop on end-of-sentence token.
+    :return: Yields a tensor representing the generated token.
+    """
+
+    # Create the config for the IREE runtime environment
+    config = ireert.Config("local-task")
```

Comment:
local-task will break on gpu usage

Reply:
cpu runtime device should be either local-sync (slow and for debugging) or local-task